diofant/integrals/risch.py

```diff
@@ -762,7 +762,9 @@ def as_poly_1t(p, t, z):
         # issue sympy/sympy#4950
         raise NotImplementedError(e)
     # Compute the negative degree parts.
-    one_t_part = Poly.from_list(reversed(one_t_part.rep.rep), *one_t_part.gens,
```

Comment:
@asmeurer, it seems this code rely on the bug of DMP.split(), which does preserve leading zeros (e.g. DMP.rep, coming from `split()`, may be like `[0, 2]` - such DMP object **is equal** to the DMP polynomial with rep `[2]`).  Is that really intentional thing?

Reply:
I have no idea. This is a relevant issue though https://github.com/sympy/sympy/issues/5131

Reply:
>  This is a relevant issue though sympy/sympy#5131

Yes, I'm aware of.  The problem is that this "hackish" solution is working due to the bug (in DMP.slice() or dmp_slice_in(), whatever).